### DISCUSSION TITLE
New PDFs with embedded Links uploaded Today - links are not clickable.  Previously uploaded PDFs w Links work.

### DISCUSSION THREAD

**Discussion by bmullan** [community/community#129664] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I've quite a few PDFs with Embedded Links that I've uploaded to many of my repo's but starting Today any New PDF w/Embedded Links, the Embedded Links are no longer Clickable.

_I did some searches and this same symptom/bug has occurred in the Past but I couldn't how they got it resolved?_

Any PDF w Links I've uploaded before Today... the Links all work.

### I _believe Github is miss-handling PDF Embedded Links:_

When I Right Click on an embedded PDF Link...  **It does NOT** ask me if I I want to **_Open Link in New Tab_**  like it should.

**_In the PDF w Embedded Links I uploaded today_** it Asks me:  **_Open Image in New Tab_** ???

**Comment by LiteBrite82** [comment]
Hi @bmullan 

Can you share an example (link to a repo / file)?

**Reply by LiteBrite82** [reply]
[I did a little experiment of my own to see if I could replicate](https://github.com/LiteBrite82/Test_repo/blob/main/PdfHyperlinkTest.pdf). 

I created a PDF with a non hyper link, link, and a hyper link. I tested as a PDF before adding it to a PR, and after the PR was approved. 

The local PDF functions as expected when viewed in a PDF viewer (chrome in my case). 

The PDF in the GitHub file browser shows proper highlighting for links, but is not able to click or follow links. There is no ability to highlight text or copy/paste either. 

However, when the file is downloaded from the repository, it functions as expected.

**Reply by bmullan** [reply]
yes that is the same thing I'm seeing.

As I stated in the previous Post.

If you Right Click on your own Test Link you too will _see Github "thinks" its an Image and not a Link_

When I Right Click on an embedded PDF Link... It does **NOT** asks me:  Open **_Link_** in New Tab like it should.
 In the PDF w Embedded Links I uploaded today it Asks me:   Open **Image** in New Tab ???

I'm using Ubuntu 24.04 and I've tried both Firefox & Chrome & its broken in both.

**Reply by LiteBrite82** [reply]
I added a not so great workaround to the readme in my test repo. 

I think the issue is that the GitHub file browser does not have native support for PDF viewing; instead it gives a snapshot of the PDF as a preview. This is an assumption, but it seems accurate as when the file is downloaded, it works as intended.

**Reply by bmullan** [reply]
@LiteBrite82 

You said:   
> *I think the issue is that the GitHub file browser does not have native support for PDF viewing; instead it gives a snapshot of the PDF as a preview.*

I ***do not*** think this is correct.   

If it was,  then why did ***the very "same" PDF*** (less 1 added line for a 3rd Video link) ***uploaded just 1 day before work?***

*So do all the other PDFs I had previously uploaded to  Github!* ... ***they all still work okay!*** 

Again, I think the telling Clue - ***is what I'd stated earlier:***

> If you ***Right Click on your own Test Link*** you see ***Github "thinks" its an Image and not a Link***   
>   
> When I Right Click on an embedded PDF Link...    
> It ***does NOT*** ask me:  Open Link in New Tab like it should.   
> In a PDF w Embedded Links I uploaded after June 25th -- it asks me:  ***Open Image in New Tab*** ?

*Do that to any "earlier" PDF* and everything works!

*Only my PDFs uploaded Jun 25th or after, are having this problem from what I can tell.*

I'm not an active programmer by any stretch today but I think somewhere in Github's own code an ***Event Handler*** got changed!

Before =>  on the ***Event of Clicking on an Embedded Link in a PDF*** the ***Action*** was => ***Display the file in some PDF Viewer***

Now =>   on the ***Event of Clicking on an Embedded Link in a PDF*** the ***Action*** now is => ***Display the file in some Image Viewer***

**Reply by bmullan** [reply]
@LiteBrite82 
also... *I just click on the your "test" Link and it still fails today.*  So if you did add some "workaround" it is not working.

**Reply by LiteBrite82** [reply]
I'll dig a bit deeper

**Comment by LiteBrite82** [comment]
💬 Your Product Feedback Has Been Submitted 🎉
Thank you for taking the time to share your insights with us! Your feedback is invaluable as we build a better GitHub experience for all our users.

Here's what you can expect moving forward ⏩
- Your input will be carefully reviewed and cataloged by members of our product teams. 
- Other users may engage with your post, sharing their own perspectives or experiences. 
- GitHub staff may reach out for further clarification or insight. 

Where to look to see what's shipping 👀
- Read the [Changelog](https://github.blog/changelog/) for real-time updates on the latest GitHub features, enhancements, and calls for feedback.
- Explore our [Product Roadmap](https://github.com/orgs/github/projects/4247), which details upcoming major releases and initiatives.

What you can do in the meantime 💻
- Upvote and comment on other user feedback Discussions that resonate with you.
- Add more information at any point! Useful details include: use cases, relevant labels, desired outcomes, and any accompanying screenshots.

As a member of the GitHub community, your participation is essential. While we can't promise that every suggestion will be implemented, we want to emphasize that your feedback is instrumental in guiding our decisions and priorities.

Thank you once again for your contribution to making GitHub even better! We're grateful for your ongoing support and collaboration in shaping the future of our platform. ⭐
